st2common/st2common/content/bootstrap.py

```diff
@@ -219,8 +257,23 @@ def register_content():
     if cfg.CONF.register.policies:
         register_policies()
 
+    if cfg.CONF.register.setup_virtualenvs:
+        setup_virtualenvs()
+
 
 def setup(argv):
+    if '--register-setup-virtualenvs' in argv:
+        # If setup-virtualenvs option is used we also want to setup config for action runner since
+        # it contains some setup virtualenv related options.
+        # Keep in mind that using "setup-virtualenv" only makes sense on nodes which run action
+        # runners.
+        # Note: This is a hack, but we can't check if that flag is provided using oslo config since
+        # we need to parse config first.
+        from st2actions import config as action_runner_config
```

Comment:
Yeah, we could do that.

As you have said, it's not ideal but we already do it for a lot of things in st2api, etc.

I can take that approach, since I personally think it's the least hacky.